lessons/specifics/poolboy.md

```diff
@@ -0,0 +1,143 @@
+---
+layout: page
+title: Poolboy
+category: specifics
+order: 1
+lang: en
+---
+
+You can easily exhaust your system resources if you allow concurrent processes to run arbitrarily. Poolboy prevents having to incur the overhead by creating a pool of workers to limit the number of concurrent processes.
+
+## Why should use Poolboy?
+
+Let's think of a specific example for a moment. You are tasked to build an application for saving user profile information to the database. If you've created a process for every user registration, you would create unbounded number of connections to your database. And at some point those connections start competing for the limited resources available in your database server. Eventually your application gets timeouts and various exceptions due to the overhead from that contention.
+
+The solution to this problem is using set of workers (processes) to limit the number of connections instead of creating a process for every user registration. Then you can easily avoid running out of your system resources.
+
+That's where Poolboy comes in. It creates a pool of workers managed by a `Supervisor` without an effort on your part to create and manage it manually. Matter of fact there are many libraries use Poolboy under the covers. For example; Postgrex's connection pool *(which in turn leveraged by Ecto when using psql)*, redis_poolex *(Redis connection pool)* are some of the popular libraries use Poolboy.
+
+{% include toc.html %}
+
+## Installation
+
+Installation is a breeze with mix. All we need to do is add Poolboy as a dependency to our `mix.exs`.  
+
+Let's create an application first:
+
+```
+$ mix new poolboy_app --sup
+$ mix deps.get
+```
+
+Add Poolboy as a dependency to our `mix.exs`.  
+
+```elixir
+defp deps do
+  [{:poolboy, "~> 1.5.1"}]
+end
+```
+
+And add Poolboy to our OTP application:
+
+```elixir
+def application do
+  [applications: [:logger, :poolboy]]
+end
+```
+
+## The configuration options
+
+We need to know little bit about the various configuration options in order to start using Poolboy.
+
+`name:` the pool name. Scope can be :local,  :global, or :via.
```

Comment:
@ayayalar could we make this list a markdown list as well as wrapping our atoms with backticks?

```markdown
+ `:max_overflow` - maximum number of workers created if pool is empty. (optional)
+ `:name` - the pool name. Scope can be `:local`, `:global`, or `:via`.
+ `:size` - maximum pool size.
+ `:strategy` - `:lifo` or `:fifo`, determines whether checked in workers should be placed first or last in the line of available workers. Default is `:lifo`. (optional)
+ `:worker_module` - the module that represents the worker.
```